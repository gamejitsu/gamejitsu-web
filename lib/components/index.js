import Button from './Button'
import ButtonSteam from './ButtonSteam'
import Card from './Card'
import Comment from './Comment'
import Layout from './Layout'
import Link from './Link'
import Navbar from './Navbar'
import Spinner from './Spinner'
import UserContext from './UserContext'

export { Button, ButtonSteam, Card, Layout, Link, Navbar, Spinner, UserContext, Comment}
