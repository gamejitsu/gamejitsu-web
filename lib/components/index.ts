import Button from "./Button"
import ButtonNew from "./ButtonNew"
import ButtonIcon from "./ButtonIcon"
import ButtonAlternative from "./ButtonAlternative"
import ButtonDark from "./ButtonDark"
import Card from "./Card"
import CardNew from "./CardNew"
import Form from "./Form"
import FormGroup from "./FormGroup"
import HeroImage from "./HeroImage"
import HeroImageSmall from "./HeroImageSmall"
import ImageButton from "./ImageButton"
import InputGroup from "./InputGroup"
import Layout from "./Layout"
import LayoutWithMenu from "./LayoutWithMenu"
import LayoutWithMenuUser from "./LayoutWithMenuUser"
import Link from "./Link"
import LinkLeftMenu from "./LinkLeftMenu"
import Navbar from "./Navbar"
import NavbarLeftMenu from "./NavbarLeftMenu"
import SkillLevelSelect from "./SkillLevelSelect"
import Spinner from "./Spinner"
import SteamButton from "./SteamButton"
import Title from "./Title"

export {
  Button,
  ButtonNew,
  ButtonIcon,
  ButtonAlternative,
  ButtonDark,
  Card,
  CardNew,
  Form,
  FormGroup,
  HeroImage,
  HeroImageSmall,
  ImageButton,
  InputGroup,
  Layout,
  LayoutWithMenu,
  LayoutWithMenuUser,
  Link,
  LinkLeftMenu,
  Navbar,
  NavbarLeftMenu,
  SkillLevelSelect,
  Spinner,
  SteamButton,
  Title
}
