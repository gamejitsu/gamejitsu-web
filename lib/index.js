import createStore from './createStore'
import theme from './theme'

export { createStore, theme }
