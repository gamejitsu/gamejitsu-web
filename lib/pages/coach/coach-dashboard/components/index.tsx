import Page from "./Page"
import CoachReviewCard from "./CoachReviewCard"
import ReviewRequestCard from "./ReviewRequestCard"

export { Page, CoachReviewCard, ReviewRequestCard }
