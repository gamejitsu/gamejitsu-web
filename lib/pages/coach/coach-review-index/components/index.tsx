import Page from "./Page"
import CoachReviewCard from "./CoachReviewCard"
import CoachDeletedReviewCard from "./CoachDeletedReviewCard"

export { Page, CoachReviewCard, CoachDeletedReviewCard }
