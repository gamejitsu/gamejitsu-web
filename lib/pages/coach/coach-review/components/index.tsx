import Page from "./Page"
import CommentBar from "./CommentBar"
import CommentFormNew from "./CommentFormNew"
import CommentList from "./CommentList"

const commentDuration = 10

export { Page, CommentBar, commentDuration, CommentFormNew, CommentList }
