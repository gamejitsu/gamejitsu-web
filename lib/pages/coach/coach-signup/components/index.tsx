import Page from "./Page"
import CoachSignUpForm from "./CoachSignUpForm"

export { Page, CoachSignUpForm }
