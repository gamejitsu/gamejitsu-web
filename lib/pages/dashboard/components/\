import React from 'react'
import { ReviewCard, Layout } from '~/components'

export default () => (
  <Layout title="Dashboard">
    Test
    <Card>Test Card</Card>
  </Layout>
)
