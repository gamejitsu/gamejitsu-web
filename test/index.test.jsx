import { mount } from 'enzyme'
import React from 'react'
import { ThemeProvider } from 'styled-components'
import { theme } from '~'
import IndexPage from '../pages/index'
import { UserContext } from '../lib/components'

const user = {
  attributes: {
    username: 'userTest'
  }
}

test('Index page should match snapshot', () => {
  const home = mount(
    <ThemeProvider theme={theme}>
      <UserContext.Provider value={{ user }}>
        <IndexPage />
      </UserContext.Provider>
    </ThemeProvider>
  )

  expect(home).toMatchSnapshot()
})
