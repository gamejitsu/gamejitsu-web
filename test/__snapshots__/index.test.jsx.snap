// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page should match snapshot 1`] = `
<ThemeProvider
  theme={
    Object {
      "activeColor": "#ffD74f",
      "backgroundColor": "#1D0D0D",
      "errorColor": "#ff665a",
      "linkColor": "#fff",
      "primaryColor": "#ADD90D",
      "secondaryColor": "#488C03",
      "textColor": "#F1F2E9",
      "textFont": "'Exo 2',sans-serif",
    }
  }
>
  <_default>
    <Layout>
      <Navbar>
        <Navbar__Container
          alignItems="center"
          color="white"
          px={2}
          py={2}
        >
          <StyledComponent
            alignItems="center"
            color="white"
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": [Function],
                "__emotion_real": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    Object {
                      "boxSizing": "border-box",
                      "margin": 0,
                      "minWidth": 0,
                    },
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    Object {
                      "display": "flex",
                    },
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                },
                "__emotion_styles": Array [
                  Object {
                    "boxSizing": "border-box",
                    "margin": 0,
                    "minWidth": 0,
                  },
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  Object {
                    "display": "flex",
                  },
                ],
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "Navbar__Container-poa0vc-0",
                  "isStatic": false,
                  "lastClassName": "cMkwtj",
                  "rules": Array [
                    "background-color:",
                    [Function],
                    ";",
                  ],
                },
                "displayName": "Navbar__Container",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "Navbar__Container-poa0vc-0",
                "target": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "__emotion_base": "div",
                  "__emotion_forwardProp": [Function],
                  "__emotion_real": [Circular],
                  "__emotion_styles": Array [
                    Object {
                      "boxSizing": "border-box",
                      "margin": 0,
                      "minWidth": 0,
                    },
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    [Function],
                    Object {
                      "display": "flex",
                    },
                  ],
                  "defaultProps": undefined,
                  "displayName": "Styled(div)",
                  "render": [Function],
                  "withComponent": [Function],
                },
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            px={2}
            py={2}
          >
            <Styled(div)
              alignItems="center"
              className="Navbar__Container-poa0vc-0 cMkwtj"
              color="white"
              px={2}
              py={2}
            >
              <div
                className="Navbar__Container-poa0vc-0 cMkwtj css-p1imuz"
              >
                <NavLink
                  fontWeight="bold"
                  href="/"
                >
                  <Styled(div)
                    m={1}
                  >
                    <div
                      className="css-1ee7t57"
                    >
                      <Link
                        href="/"
                      >
                        <Link
                          href="/"
                          prefetch={true}
                        >
                          <Link__Content
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Link__Content-j1roec-0",
                                    "isStatic": false,
                                    "lastClassName": "senXN",
                                    "rules": Array [
                                      "color:",
                                      [Function],
                                      ";cursor:pointer;",
                                    ],
                                  },
                                  "displayName": "Link__Content",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Link__Content-j1roec-0",
                                  "target": "a",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={[Function]}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="Link__Content-j1roec-0 senXN"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                Gamejitsu
                              </a>
                            </StyledComponent>
                          </Link__Content>
                        </Link>
                      </Link>
                    </div>
                  </Styled(div)>
                </NavLink>
                <NavLink
                  href="/dashboard"
                >
                  <Styled(div)
                    m={1}
                  >
                    <div
                      className="css-1ee7t57"
                    >
                      <Link
                        href="/dashboard"
                      >
                        <Link
                          href="/dashboard"
                          prefetch={true}
                        >
                          <Link__Content
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Link__Content-j1roec-0",
                                    "isStatic": false,
                                    "lastClassName": "senXN",
                                    "rules": Array [
                                      "color:",
                                      [Function],
                                      ";cursor:pointer;",
                                    ],
                                  },
                                  "displayName": "Link__Content",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Link__Content-j1roec-0",
                                  "target": "a",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={[Function]}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="Link__Content-j1roec-0 senXN"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                Dashboard
                              </a>
                            </StyledComponent>
                          </Link__Content>
                        </Link>
                      </Link>
                    </div>
                  </Styled(div)>
                </NavLink>
                <Styled(div)
                  mx="auto"
                >
                  <div
                    className="css-1ntkiwo"
                  />
                </Styled(div)>
                <NavLink
                  href="/"
                  key="username"
                >
                  <Styled(div)
                    m={1}
                  >
                    <div
                      className="css-1ee7t57"
                    >
                      <Link
                        href="/"
                      >
                        <Link
                          href="/"
                          prefetch={true}
                        >
                          <Link__Content
                            onClick={[Function]}
                            onMouseEnter={[Function]}
                          >
                            <StyledComponent
                              forwardedComponent={
                                Object {
                                  "$$typeof": Symbol(react.forward_ref),
                                  "attrs": Array [],
                                  "componentStyle": ComponentStyle {
                                    "componentId": "Link__Content-j1roec-0",
                                    "isStatic": false,
                                    "lastClassName": "senXN",
                                    "rules": Array [
                                      "color:",
                                      [Function],
                                      ";cursor:pointer;",
                                    ],
                                  },
                                  "displayName": "Link__Content",
                                  "foldedComponentIds": Array [],
                                  "render": [Function],
                                  "styledComponentId": "Link__Content-j1roec-0",
                                  "target": "a",
                                  "toString": [Function],
                                  "warnTooManyClasses": [Function],
                                  "withComponent": [Function],
                                }
                              }
                              forwardedRef={[Function]}
                              onClick={[Function]}
                              onMouseEnter={[Function]}
                            >
                              <a
                                className="Link__Content-j1roec-0 senXN"
                                onClick={[Function]}
                                onMouseEnter={[Function]}
                              >
                                userTest
                              </a>
                            </StyledComponent>
                          </Link__Content>
                        </Link>
                      </Link>
                    </div>
                  </Styled(div)>
                </NavLink>
                <Button
                  key="logout"
                  onClick={[Function]}
                  text="Logout"
                >
                  <Button__Content
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <StyledComponent
                      disabled={false}
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "Button__Content-sc-1w7q4jt-0",
                            "isStatic": false,
                            "lastClassName": "pjMHR",
                            "rules": Array [
                              "background:linear-gradient( to bottom,",
                              [Function],
                              ",",
                              [Function],
                              " );padding:12px 30px;font-size:15px;outline:0;border:0;cursor:pointer;",
                            ],
                          },
                          "displayName": "Button__Content",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "Button__Content-sc-1w7q4jt-0",
                          "target": "button",
                          "toString": [Function],
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      onClick={[Function]}
                      type="button"
                    >
                      <button
                        className="Button__Content-sc-1w7q4jt-0 pjMHR"
                        disabled={false}
                        onClick={[Function]}
                        type="button"
                      >
                        Logout
                      </button>
                    </StyledComponent>
                  </Button__Content>
                </Button>
              </div>
            </Styled(div)>
          </StyledComponent>
        </Navbar__Container>
      </Navbar>
      <Layout__Container
        mx="auto"
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "__emotion_base": "div",
              "__emotion_forwardProp": [Function],
              "__emotion_real": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": [Function],
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  Object {
                    "boxSizing": "border-box",
                    "margin": 0,
                    "minWidth": 0,
                  },
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "defaultProps": undefined,
                "displayName": "Styled(div)",
                "render": [Function],
                "withComponent": [Function],
              },
              "__emotion_styles": Array [
                Object {
                  "boxSizing": "border-box",
                  "margin": 0,
                  "minWidth": 0,
                },
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "Layout__Container-sc-16wg312-0",
                "isStatic": true,
                "lastClassName": "hEWBOx",
                "rules": Array [
                  "max-width:1024px;",
                ],
              },
              "displayName": "Layout__Container",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "Layout__Container-sc-16wg312-0",
              "target": Object {
                "$$typeof": Symbol(react.forward_ref),
                "__emotion_base": "div",
                "__emotion_forwardProp": [Function],
                "__emotion_real": [Circular],
                "__emotion_styles": Array [
                  Object {
                    "boxSizing": "border-box",
                    "margin": 0,
                    "minWidth": 0,
                  },
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                  [Function],
                ],
                "defaultProps": undefined,
                "displayName": "Styled(div)",
                "render": [Function],
                "withComponent": [Function],
              },
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          mx="auto"
        >
          <Styled(div)
            className="Layout__Container-sc-16wg312-0 hEWBOx"
            mx="auto"
          >
            <div
              className="Layout__Container-sc-16wg312-0 hEWBOx css-1ntkiwo"
            >
              <Head>
                <Component
                  handleStateChange={null}
                  inAmpMode={false}
                  reduceComponentsToState={[Function]}
                />
              </Head>
            </div>
          </Styled(div)>
        </StyledComponent>
      </Layout__Container>
    </Layout>
  </_default>
</ThemeProvider>
`;
